pkg/sql/sem/tree/datum.go

```diff
@@ -901,6 +927,11 @@ func NewDCollatedString(
 // AmbiguousFormat implements the Datum interface.
 func (*DCollatedString) AmbiguousFormat() bool { return false }
 
+// AsJSON implements transformation to JSON
+func (d *DCollatedString) AsJSON() (json.JSON, error) {
+	return json.MakeJSONFromString(""), nil
```

Comment:
This needs to work, but I think you can just use the string value, collation just affects sorting within SQL so I don't think there's anything special that needs to happen here.